link-grammar/disjunct-utils.c

```diff
@@ -371,13 +371,13 @@ Disjunct *eliminate_duplicate_disjuncts(Disjunct *dw)
 				gword_set_union(dx->originating_gword, d->originating_gword);
 
 			count++;
-			*dd = d->next; /* NEXT - set current disjunct to the next one. */
+			prev->next = d->next;
```

Comment:
prev will be null if the very first disjuncts has a non null dx ... maybe you want to initialize `prev=dw`;` instead?

Reply:
Note what I wrote in the commit message.
I guess a comment would be a good idea:
``` c
    350         Disjunct *prev = NULL;  /* The first disjunct is never eliminated. */
```

Reply:
The comment would be good; I rarely read commit messages, unless there's a problem.